src/core/Form/Combobox/Combobox/Combobox.test.tsx

```diff
@@ -0,0 +1,364 @@
+      expect(chips.length).toEqual(1);
+      const disabledChip = chips[0];
+      expect(disabledChip).toHaveTextContent('Powersaw');
+    });
+  });
+
+  test('onItemSelect: called with uniqueItem id, when clicking non-disabled Chip', async () => {
+    await act(async () => {
+      const onItemSelect = (uniqueItemId: string) => console.log(uniqueItemId);
+      const onItemSelectSpy = jest.spyOn(console, 'log');
+      const { container } = render(
+        <Combobox
+          labelText="Combobox"
+          items={tools}
+          chipListVisible={true}
+          chipActionLabel="Remove"
+          removeAllButtonLabel="Remove all selections"
+          visualPlaceholder="Choose your tool(s)"
+          emptyItemsLabel="No items"
+          defaultSelectedItems={defaultSelectedTools}
+          onItemSelect={onItemSelect}
+        />,
+      );
+      const hammerChip = container.querySelectorAll('.fi-chip')[0];
+      await act(async () => {
+        fireEvent.click(hammerChip, {});
+      });
+      expect(onItemSelectSpy).toBeCalledWith('h9823523');
+    });
+  });
+
+  it('second Chip should be disabled', () => {
+    const { container } = render(BasicCombobox);
+    const disabledChip = container.querySelectorAll('.fi-chip')[1];
+    expect(disabledChip).toHaveTextContent('Powersaw');
+    expect(disabledChip).toHaveClass('fi-chip--disabled');
+    expect(disabledChip).toHaveAttribute('disabled');
+  });
+
+  it('should remove all non-disabled Chips when pressing "Remove all" button', async () => {
+    await act(async () => {
+      const { container } = render(BasicCombobox);
+      expect(container.querySelectorAll('.fi-chip').length).toEqual(2);
+      const removeAllButton = container.querySelectorAll(
+        '.fi-combobox_removeAllButton',
+      )[0];
+      await act(async () => {
+        fireEvent.click(removeAllButton, {});
+      });
+      const chips = container.querySelectorAll('.fi-chip');
+      expect(chips.length).toEqual(1);
+    });
+  });
+
+  test('onRemoveAll: should be called when pressing "Remove all" button', async () => {
+    await act(async () => {
+      const mockOnRemoveAll = jest.fn();
+      const { container } = render(
+        <Combobox
+          labelText="Combobox"
+          items={tools}
+          chipListVisible={true}
+          chipActionLabel="Remove"
+          removeAllButtonLabel="Remove all selections"
+          visualPlaceholder="Choose your tool(s)"
+          emptyItemsLabel="No items"
+          defaultSelectedItems={defaultSelectedTools}
+          onRemoveAll={mockOnRemoveAll}
+        />,
+      );
+      const removeAllButton = container.querySelectorAll(
+        '.fi-combobox_removeAllButton',
+      )[0];
+      await act(async () => {
+        fireEvent.click(removeAllButton, {});
+      });
+      expect(mockOnRemoveAll).toBeCalledTimes(1);
+    });
+  });
+});
+
+describe('Controlled', () => {
+  it('has the controlled items as selected + disabled', async () => {
```

Comment:
Maybe add a test for checking that controlled state prevents selecting / deselecting if controlled state is not changed?

Reply:
Good idea 👍

Reply:
Added a test that checks that item couldn't be removed when it was controlled, using `selectedItems` prop.